File: algorithms/python/minimumOperationstoMakeArrayEqual/minimumOperationstoMakeArrayEqual.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/minimum-operations-to-make-array-equal/
# Author: Miao Zhang
# Date:   2021-05-15

class Solution:
    def minOperations(self, n: int) -> int:
        res = 0
        avg = n
        for i in range(n // 2):
            num = 2 * i + 1
            if num < n:
                res += n - num
        return res

File: algorithms/cpp/minimumOperationstoMakeArrayEqual/minimumOperationstoMakeArrayEqual.cpp
// Source: https://leetcode.com/problems/minimum-operations-to-make-array-equal/
// Author: Miao Zhang
// Date:   2021-05-15

class Solution {
public:
    int minOperations(int n) {
        int res = 0;
        int avg = n;
        for (int i = 0; i < n; i++) {
            int num = 2 * i + 1;
            if (num < n) {
                res += n - num;
            }
        }
        return res;
    }
};
